go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -264,6 +264,17 @@ func (c *Controller) Run(stopCh <-chan struct{}, wg *sync.WaitGroup, threads int
 			if err = c.iptablesManager.EnsureRule(utiliptables.TableMangle, utiliptables.ChainPrerouting, utiliptables.ProtocolIPv4, iptSaveMarkRule); err != nil {
 				return fmt.Errorf("failed to create rule in chain %s to save pkt marking: %v", utiliptables.ChainPrerouting, err)
 			}
+
+			// If dst is a node IP, use main routing table and skip EIP routing tables
+			if err = c.ruleManager.Add(getNodeIPFwMarkIPRule(netlink.FAMILY_V4)); err != nil {
+				return fmt.Errorf("failed to create IP rule for node IPs: %v", err)
```

Comment:
nit add family in the error message and maybe even the nodeIPs..
i think this error and the one for v6 will look exactly the same causing debugging confusion

Reply:
Added the family to distinguish between v4 and v6. No need to add nodeIPs as they do not really exist in this context :) The rule doesn't contain any IPs.